src/Compilers/CSharp/Test/Syntax/Parsing/LambdaParameterParsingTests.cs

```diff
@@ -588,7 +588,7 @@ public void HangingLambdaParsing_Bug14167()
         [Fact]
         public void TestLambdaWithNullValidation()
         {
-            UsingDeclaration("Func<string, string> func1 = x! => x + \"1\";", options: TestOptions.RegularPreview);
+            UsingDeclaration("Func<string, string> func1 = x!! => x + \"1\";", options: TestOptions.RegularPreview);
```

Comment:
We need a set of rolling parsing spaces tests. ie, test all of these cases:
```cs
x !!=> ...;
x! !=> ....;
x!! => ...;
x!!= > ...;
```

Same for parameter parsing as well.

Reply:
Or this comment.

---
In reply to: [466678757](https://github.com/dotnet/roslyn/pull/46520#discussion_r466678757) [](ancestors = 466678757)

Reply:
Added parsing tests!